### PR TITLE
Make prepend (and append) filters available in ActiveAdmin.setup

### LINKED ISSUE #3872: Want to use prepend_before_filter in ActiveAdmin.setup
I have wanted to add an authentication filter on the top of everything, even skip login redirection.

In a regular Rails app, you can achieve this by using [`prepend_before_filter`](http://apidock.com/rails/v3.2.13/AbstractController/Callbacks/ClassMethods/prepend_before_filter). However, I found that current ActiveAdmin just delegate some part of filters, thus I couldn't use `prepend_before_filter`.

I've just added the missing methods [here](https://github.com/activeadmin/activeadmin/blob/master/lib/active_admin/application.rb#L219) and it works fine. I'll submit a pullreq later.

### PR DESCRIPTION
**Pull request by tnj**
This fixes #3872. :beers:

### CODE REVIEW

**Comment by timoschilling** [comment]
1. I missing the specs.
2. We should support Rails 4 `before_action` too.
3. We should build a more generic code like:
```ruby
prefixes = [nil, "skip", "append", "prepend"]
positions = ["before", "around", "after"]
subfixes = ["filter", "action"]
prefixes.each_with_object([]){|prefix, stack| positions.each{|position| subfixes.each{|subfix| stack << [prefix, position, subfix].compact.join("_") } } }

```

**Comment by tnj** [comment]
Hey @timoschilling,

Rather than generating method names from hardcoded values, enumerating the public instance methods from [`AbstractController::Callbacks::ClassMethods`](http://apidock.com/rails/v4.1.8/AbstractController/Callbacks/ClassMethods) could be much more robust against the future changes on Rails.

I've revised the code in this manner, and added tests based on your example code to ensure the existence of methods what we are expecting regardless the Rails modification.

File: spec/unit/controller_filters_spec.rb [hunk 1]
```diff
@@ -12,33 +12,29 @@
...
+        end
+      end
+    end
+  )
+
+  expected_actions.each do |action|
+    it action do
+      controllers.each { |c| expect(c).to receive(action).and_return(true) }
+      application.public_send action, [:my_filter, { only: :show }]
+    end
```
Comment (timoschilling) [inline spec/unit/controller_filters_spec.rb]:
```ruby
it action do
  args = [:my_filter, { only: :show }]
  controllers.each { |c| expect(c).to receive(action).with(args) } # I think `.and_return(true)` is not needed
  application.public_send action, args
end
```

Comment (tnj) [inline spec/unit/controller_filters_spec.rb]: :+1: Agreed.

**Comment by timoschilling** [comment]
Good work!
Please apply my suggested chance and squash your commits.

**Comment by tnj** [comment]
Thanks :+1:  I squashed the commits.

**Pull request merged**